File: src/lessons/welcome/slug/SlugTrackingEntity.java
package lessons.welcome.slug;

import java.awt.Color;


public class SlugTrackingEntity extends jlm.universe.bugglequest.SimpleBuggle {

	@Override
	public void run() {
		hunt(); 
	}

	public void hunt() {
		while (! isOverBaggle()) {
			if (isFacingTrail()) {
				brushDown();
				forward();
				brushUp();
			} else {
				turnLeft();
			}
		}
		pickupBaggle();
	}

	/* BEGIN TEMPLATE */
	boolean isFacingTrail() {
		// Write your code here
		/* BEGIN SOLUTION */
		if (isFacingWall())
			return false;
		forward();
		boolean res = getGroundColor().equals(Color.green); 
		backward();
		return res;
		/* END SOLUTION */
	}		
	/* END TEMPLATE */

}

File: resources/exercises/methods/slug/SlugTrackingEntity.py
import java.awt.Color as Color

# BEGIN TEMPLATE
def isFacingTrail():
   # write your code here
   # BEGIN SOLUTION
   if isFacingWall():
      return False
   else:
	  forward()
	  res = (getGroundColor() == Color.green)
	  backward()
	  return res
   # END SOLUTION
# END TEMPLATE

def hunt():
   while not isOverBaggle():
      brushUp()
      if isFacingTrail():
         brushDown()
         forward()
         brushUp()
      else:
	     left()
   pickupBaggle()

hunt()

